src/docfx/serve/LanguageServerBuilder.cs

```diff
@@ -30,28 +31,47 @@ internal class LanguageServerBuilder
             _errorList = new();
             _languageServerPackage = languageServerPackage;
             _builder = new(_errorList, languageServerPackage.BasePath, options, _languageServerPackage);
-            _buildSemaphore = new(0);
+            _buildChannel = Channel.CreateUnbounded<bool>();
             _ = StartAsync();
         }
 
         public void QueueBuild()
```

Comment:
will different actions have different denounce timeout? E.g. textChange is 1s, but textOpen is immediate?

Reply:
It is nice to have and we can re-consider about it in the future?